src/common/utils/thumbnailUrlUtils.ts

```diff
@@ -0,0 +1,22 @@
+import { IVideoInfo } from "../interfaces/IVideoInfo";
+
+type VideoProvider = 'youtube' | 'vimeo';
+
+const youtubeRegex = /(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/;
+const vimeoRegex = /(?:https?:\/\/)?(?:www\.)?(?:player\.)?vimeo\.com\/(?:video\/)?(\d+)/;
+
+/**
+ * Accepts url from IVideoData as an arg and returns a tuple of the VideoProvider and the Id
+ */
+export function getVideoProviderAndId(videoInfo: IVideoInfo): [VideoProvider | 'unknown', string | null] {
```

Comment:
🔧 This typing is technically correct. However, I think we can make it even more clear and give the compiler even more info.  Something like this:

```suggestion
export function getVideoProviderAndId(videoInfo: IVideoInfo): [VideoProvider, string] | ['unknown', null] {
```

In this way, the compiler knows that if the `provider` is `'unknown'`, then the `id` is `null`. And vice versa, if the `provider` is not `'unknown'`, then the `id` is a non-nullable `string`